// Proto file for auth service

syntax = "proto3";

package auth;

import "user/protos/user.proto";

service Auth {
    rpc Login(LoginRequest) returns (LoginResponse) {}
    rpc Signup(SignupRequest) returns (SignupResponse) {}
}

// Request - responses --------------------------------------------------------------------

message LoginRequest {
    string email = 1;
    string password = 2;
}
message LoginResponse {
    int64 user_id = 1;
    string access_token = 2;
    string refresh_token = 3;
}

message SignupRequest {
    user.UserSchema user = 1;
}
message SignupResponse {
    string status = 1;
}
